File: Valid_Palindrome/valid_palindrome.js
/**
 * @param {string} s - The string to check
 * @returns {boolean} Return whether palindrome string is valid
 */
var isPalindrome = function (s) {
	const cleaned = s.toLowerCase().replace(/[^a-z0-9]/g, "");

	let i = 0;
	let j = cleaned.length - 1;

	while (i < j) {
		if (cleaned[i] !== cleaned[j]) {
			return false;
		}

		i++;
		j--;
	}

	return true;
};

File: Valid_Palindrome/valid_palindrome.dart
class Solution {
  bool isPalindrome(String s) {
    final String cleaned = s.toLowerCase().replaceAll(RegExp(r'[^a-z0-9]'), '');

    int i = 0;
    int j = cleaned.length - 1;

    while (i < j) {
      if (cleaned[i] != cleaned[j]) {
        return false;
      }
      i++;
      j--;
    }

    return true;
  }
}
